shared/chat/inbox/row/chat-filter-row/container.tsx

```diff
@@ -14,45 +14,40 @@ type OwnProps = {
   query: string
 }
 
-const mapStateToProps = (state, ownProps: OwnProps) => {
-  return {
-    filter: ownProps.query,
-    isSearching: !!state.chat2.inboxSearch,
-  }
+function KeyHandler<T>(t: T): T {
+  return isMobile ? t : require('../../../../util/key-handler.desktop').default
 }
 
-const mapDispatchToProps = dispatch => ({
-  _onHotkey: (cmd: string) => {
-    if (cmd.endsWith('+n')) {
-      dispatch(appendNewChatBuilder())
-    }
-  },
-  onBack: () => dispatch(RouteTreeGen.createNavigateUp()),
-  onStartSearch: () => dispatch(Chat2Gen.createToggleInboxSearch({enabled: true})),
-  onStopSearch: () => dispatch(Chat2Gen.createToggleInboxSearch({enabled: false})),
-})
-
-const mergeProps = (stateProps, dispatchProps, ownProps) => ({
-  _onHotkey: dispatchProps._onHotkey,
-  filter: stateProps.filter,
-  hotkeys: isDarwin ? ['command+n'] : ['ctrl+n'],
-  isSearching: stateProps.isSearching,
-  onBack: dispatchProps.onBack,
-  onEnsureSelection: ownProps.onEnsureSelection,
-  onNewChat: ownProps.onNewChat,
-  onSelectDown: ownProps.onSelectDown,
-  onSelectUp: ownProps.onSelectUp,
-  onSetFilter: ownProps.onQueryChanged,
-  onStartSearch: dispatchProps.onStartSearch,
-  onStopSearch: dispatchProps.onStopSearch,
-})
+const Component = isMobile ? ConversationFilterInput : KeyHandler(ConversationFilterInput)
```

Comment:
Why do the `isMobile` twice (here and in the definition of `KeyHandler`)? Why put `KeyHandler` in the weird templated function above?